plugin/trino-tpch/src/main/java/io/trino/plugin/tpch/TpchSplit.java

```diff
@@ -93,21 +92,15 @@ public List<HostAddress> getAddresses()
     @Override
     public boolean equals(Object obj)
     {
-        if (this == obj) {
-            return true;
-        }
-        if ((obj == null) || (getClass() != obj.getClass())) {
-            return false;
-        }
-        TpchSplit other = (TpchSplit) obj;
-        return Objects.equals(this.totalParts, other.totalParts) &&
-                Objects.equals(this.partNumber, other.partNumber);
+        //  There is no equality contract for ConnectorSplit and engine should not call this method.
+        throw new UnsupportedOperationException();
```

Comment:
Remove instead? Or if we want to be safer maybe we could add this to the ConnectorSplit as a default implementation?